process_report/process_report.py

```diff
@@ -112,21 +228,80 @@ def remove_billables(dataframe, pi, projects, output_file):
 
     So this *keeps* the projects/pis that should not be billed.
     """
-    filtered_dataframe = dataframe[dataframe['Manager (PI)'].isin(pi) | dataframe['Project - Allocation'].isin(projects)]
+    filtered_dataframe = dataframe[dataframe[MANAGER_PI_FIELD].isin(pi) | dataframe[PROJECT_FIELD].isin(projects)]
     filtered_dataframe.to_csv(output_file, index=False)
 
+
 def export_pi_billables(dataframe: pandas.DataFrame, output_folder):
     if not os.path.exists(output_folder):
         os.mkdir(output_folder)
 
-    invoice_month = dataframe['Invoice Month'].iat[0]
-    pi_list = dataframe['Manager (PI)'].unique()
+    invoice_month = dataframe[INVOICE_MONTH_FIELD].iat[0]
+    pi_list = dataframe[MANAGER_PI_FIELD].unique()
 
     for pi in pi_list:
-        pi_projects = dataframe[dataframe['Manager (PI)'] == pi]
-        pi_instituition = pi_projects['Institution'].iat[0]
+        if pi != pi: continue
+        pi_projects = dataframe[dataframe[MANAGER_PI_FIELD] == pi]
+        pi_instituition = pi_projects[INSTITUTION_FIELD].iat[0]
         pi_projects.to_csv(output_folder + f"/{pi_instituition}_{pi}_{invoice_month}.csv")
-        
+
+
+def add_credits(dataframe : pandas.DataFrame):
+    """Adds credits to PIs depending on different criterions"""
+    dataframe.insert(dataframe.columns.get_loc(COST_FIELD) + 1, CREDIT_FIELD, 0.0)
+    dataframe.insert(dataframe.columns.get_loc(CREDIT_FIELD) + 1, CREDIT_CODE_FIELD, None)
+    dataframe.insert(dataframe.columns.get_loc(CREDIT_CODE_FIELD) + 1, BALANCE_FIELD, 0.0)
+
+    # Apply credits
+    for credit_func in applied_credits:
+        dataframe = credit_func(dataframe)
+
+    return dataframe
+
+
+def add_institution(dataframe: pandas.DataFrame):
+    """Determine the PI's institution name, logging any PI whose institution cannot be determined"""
+    for i, row in dataframe.iterrows():
+        pi_name = row[MANAGER_PI_FIELD]
+        if pi_name != pi_name: print(f"Project {row[PROJECT_FIELD]} has no PI") # Nan check
+        else: dataframe.at[i, INSTITUTION_FIELD] = get_institution_from_pi(pi_name)
+
+    return dataframe
+
+
+def export_HU_only(dataframe, output_file):
+    HU_projects = dataframe[dataframe[INSTITUTION_FIELD] == 'Harvard University']
+    HU_projects.to_csv(output_file)
+
+
+def export_HU_BU(dataframe, output_file):
+    HU_BU_projects = dataframe[(dataframe[INSTITUTION_FIELD] == 'Harvard University') | 
+                               (dataframe[INSTITUTION_FIELD] == 'Boston University')]
+    HU_BU_projects.to_csv(output_file)        
+
+
+def export_lenovo(dataframe: pandas.DataFrame, output_file=None):
+
+    invoice_date = dataframe[INVOICE_MONTH_FIELD].iat[0].split('-')
+    invoice_year = invoice_date[0]
+    invoice_month = invoice_date[1]
+    lenovo_file_name = output_file or f'Lenovo_{invoice_year}_{invoice_month}.csv'
+
+    LENOVO_SU_TYPES = ['OpenShift GPUA100SXM4', 'OpenStack GPUA100SXM4']
+    SU_CHARGE_MULTIPLIER = 1
```

Comment:
@knikolla are we flagging rows that would be useful to add to [this](https://github.com/nerc-project/operations/issues/420) repo once it is ready?